src/Host/Client/Impl/BrokerServices/RemoteUriWebService.cs

```diff
@@ -5,14 +5,180 @@
 using System.Net.Http;
 using System.Threading.Tasks;
 using System.IO;
+using System.Net;
+using System.Collections.Generic;
+using Microsoft.Common.Core;
+using System.Threading;
+using System.Collections.Specialized;
+using System.Text;
 
 namespace Microsoft.R.Host.Client.BrokerServices {
-    public class RemoteUriWebService : WebService, IRemoteUriWebService {
-        public RemoteUriWebService(HttpClient httpClient) : base(httpClient) {
+    public class RemoteUriWebService : IRemoteUriWebService {
+        public RemoteUriWebService(string baseUri) {
+            PostUri = new Uri(new Uri(baseUri),"/remoteuri");
         }
+        private Uri PostUri { get; }
 
-        private static readonly Uri postUri = new Uri("/remoteuri", UriKind.Relative);
+        public async Task GetResponseAsync(HttpListenerContext context, string localBaseUrl, string remoteBaseUrl, CancellationToken ct) {
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(PostUri);
+            request.Method = context.Request.HttpMethod;
+            SetRequestHeaders(request, context.Request.Headers, localBaseUrl, remoteBaseUrl);
 
-        public Task<Stream> PostAsync(Stream request) => HttpPostAsync(postUri, request);
+            // Add RTVS headers
+            request.Headers.Add("x-rtvs-url", GetRemoteUrl(context.Request.Url, remoteBaseUrl));
+
+            if(context.Request.InputStream.CanSeek && context.Request.InputStream.Length > 0) {
+                Stream reqStream = await request.GetRequestStreamAsync();
+                await context.Request.InputStream.CopyToAsync(reqStream);
+                await reqStream.FlushAsync();
+            }
+            
+            HttpWebResponse response =  (HttpWebResponse)await request.GetResponseAsync();
+
+            SetResponseHeaders(response, context.Response, localBaseUrl, remoteBaseUrl);
+            Stream respStream = response.GetResponseStream();
+
+            await respStream.CopyToAsync(context.Response.OutputStream);
+            await context.Response.OutputStream.FlushAsync();
+            context.Response.OutputStream.Close();
+        }
+
+        private string GetRemoteUrl(Uri url, string remoteBase) {
+            Uri remote = new Uri($"http://{remoteBase}");
+            UriBuilder ub = new UriBuilder(url);
+            ub.Host = remote.Host;
+            ub.Port = remote.Port;
+            return ub.Uri.ToString();
+        }
+
+        private static long GetLong(string value) {
+            long ret;
+            if (long.TryParse(value, out ret)) {
+                return ret;
+            }
+            return 0;
+        }
+
+        private static DateTime GetDateTime(string value) {
+            DateTime ret;
+            if (DateTime.TryParse(value, out ret)) {
```

Comment:
Same thing here, no need for `if`.

Also, these conversion methods should probably be named something like `ToDateTimeOrDefault` (for consistency with existing `ToDateTime` and `TryToDateTime`, and LINQ methods like `FirstOrDefault`). And probably become public extension methods somewhere in `Microsoft.Common.Core`.